File: Projeto-1.py
opção = 0
carro=0
lista=[]


class Carro:

    def __init__(self,marca,modelo,ano,placa):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.placa = placa


    def __str__(self):

        '''Retorna os atributos do objeto em formato de dicionário'''

        return('=-=' * 10) + '\n' +  '\n' + '\n'.join(('{} = {}'
        .format(item, self.__dict__[item]) for item in self.__dict__)) + '\n' 


def menu():

    '''Imprime o menu ao usuário'''

    print('CATÁLOGO DE VEÍCULOS')
    print('[1] CADASTRAR NOVO VEICULO', f"({vagas_livres()} vagas)")
    print('[2] LISTAR VEICULOS')
    print('[3] PESQUISAR UM VEICULO')
    print('[4] MOSTRAR VEICULO A PARTIR DO ANO')
    print('[5] SAIR')


def user_input():     

    '''Recebe dados do usuário e atribui ás variaveis da classe Carro()'''

    marca = input('Digite a marca:').lower()
    modelo = input('Digite o modelo:').lower()
    ano = input('Digite o ano: ').lower()
    placa = input('Digite a placa:').lower() 
    global carro
    carro = Carro(marca,modelo,ano,placa)
    lista.append(carro)
 

def listar():

    '''Imprime os valores da lista para o usuário quando solicitado'''

    for l in lista:
        print(l)


def busca_modelo():

    '''Realiza busca de carros por modelo'''

    modelo = input ('Digite o modelo a ser pesquisado: ').lower()
    for l in lista:
        if l.modelo == modelo:
            print(l)
        else:
            print('Modelo não encontrado!')
            

def ano_busca():

    '''Realiza busca de carros a partir de um ano'''

    ano = input ('Digite o ano a ser pesquisado: ').lower()

    for l in lista:
        if l.ano >= ano:
            print(l)
        else:
            print('Não foram encontrados veiculos')
            continue


def vagas_livres():

    '''Retorna a quantidade de vagas restantes para cadastro de carros'''

    return 10-len(lista)        


while opção != 5:

    menu()

    opção = input('Digite sua opção: ')

    if opção == '1':

        if len(lista) <=10:
            user_input()

        else:
            print('O limite de cadastros é 10!')

    elif opção == '2':
        listar()

    elif opção == '3':
        busca_modelo()

    elif opção == '4':
        ano_busca()

    elif opção == '5':
        print('Finalizando...')

    else:
        print('Opção inválida tente novamente!')
    print('=-=' * 10)
    
print('Fim do programa!')

File: Projeto1.cpp
#include <iostream>
#include <string.h>
#include <conio.h>
#include <stdio.h>
#include <stdlib.h>
#include <locale.h>


typedef struct carro

{

int cod;

char marca [20];

char modelo [20];

int ano;

char placa[10];

}carro;

carro car[10];

void menu();

void inicializar();

void imprimir();

void cadastrar();

int main(int argc, char** argv) {

setlocale(LC_ALL,"portuguese");

int i;

int op;

int ano;

int z;

char modelo[20];

z=0;

ano=0;

i=0;

car[i].cod = 0;

strcpy(car[i].marca,"NULL");

strcpy(car[i].modelo,"NULL");

strcpy(car[i].placa,"NULL");

car[i].ano = 0;

do
{

printf("\n Opções:");

printf("\n 1. CADASTRAR VEÍCULO");

printf("\n 2. LISTAR VEÍCULOS");

printf("\n 3. PESQUISAR UM MODELO");

printf("\n 4. MOSTRAR VEÍCULOS A PARTIR DE UM ANO");

printf("\n 5. SAIR");

printf("\n \n DIGITE A OPÇÃO: ");

scanf("%d",&op);

switch(op)

{

case 1:

if (car[i].cod<=9)

{

car[i].cod=i;

printf("\n Digite a marca do veículo:");

scanf("%s",&car[i].marca);

printf("\n Digite o modelo do veículo:");

scanf("%s",&car[i].modelo);

printf("\n Digite a placa do veículo:");

scanf("%s",&car[i].placa);

printf("\n Digite o ano do veículo:");

scanf("%d",&car[i].ano);

i=i+1;



}



else



{

printf("\n Limite de cadastro é 10!");

}



break;



case 2:

for(int x=0; x<i; x++)



{

printf("\n A marca do veículo é: %s",car[x].marca);

printf("\n O modelo do veículo é: %s",car[x].modelo);

printf("\n O ano do veículo é: %d",car[x].ano);

printf("\n A placa do veículo é: %s",car[x].placa);

printf("\n\n =================================== \n \n");

}



break;



case 3:

printf("\n INSIRA O MODELO A SER PESQUISADO:");

scanf("%s",&modelo);

for (int x=0; x<i; x++)



{

if (strcmp(car[x].modelo,strlwr(modelo)) == 0)

{

printf("\n A marca do veículo é: %s",car[x].marca);

printf("\n O modelo do veículo é: %s",car[x].modelo);

printf("\n O ano do veículo é: %d",car[x].ano);

printf("\n A placa do veículo é: %s",car[x].placa);

printf("\n\n =================================== \n \n");



z++;

}



}



if (z==0)



{

printf("\n NÃO FOI ENCONTRADO O MODELO %s \n\n",modelo);

}



else



{

z=0;

}



break;



case 4:

printf("\n INSIRA O ANO A SER PESQUISADO:");

scanf("%d",&ano);



for (int x=0; x<i; x++)

{

if (car[x].ano >= ano)



{

printf("\n A marca do veículo é: %s",car[x].marca);

printf("\n O modelo do veículo é: %s",car[x].modelo);

printf("\n O ano do veículo é: %d",car[x].ano);

printf("\n A placa do veículo é: %s",car[x].placa);

printf("\n\n =================================== \n \n");



z++;

}

}



if (z==0)

{

printf("\n NÃO FOI ENCONTRADO VEÍCULOS ACIMA DE %d \n\n",ano);

}

else


{

z=0;

}

break;

default:

printf("\n Opção não existe");



break;
}

}

while (op!=5 && i<=9);

if (i==9)

{

printf("\n \n LIMITE DE VEÍCULOS ALCANÇADO! \n \n \n");

}

for (int y=0; y<i; y++)

{

printf("\n A marca do veículo é: %s",car[y].marca);

printf("\n O modelo do veículo é: %s",car[y].modelo);

printf("\n O ano do veículo é: %d",car[y].ano);

printf("\n A placa do veículo é: %s",car[y].placa);

printf("\n\n =================================== \n \n");

}

return 0;

}
